render/render.js

```diff
@@ -91,12 +93,17 @@ module.exports = function($window) {
 		return element
 	}
 	function createComponent(vnode, hooks, ns) {
-		// For object literals since `Vnode()` always sets the `state` field.
-		if (!vnode.state) vnode.state = {}
-		assign(vnode.state, vnode.tag)
+		if (typeof vnode.tag === "function") {
+			vnode.state = null
+			vnode.state = vnode.tag.prototype instanceof Component ? new vnode.tag(vnode) : vnode.tag(vnode)
+		} else {
+			// For object literals since `Vnode()` always sets the `state` field.
+			if (!vnode.state) vnode.state = {}
+			assign(vnode.state, vnode.tag)
```

Comment:
It took me seeing this line to finally realize that you effectively changed `vnode.tag` to `vnode.state`, and that this is how you did the PR.